code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -0,0 +1,207 @@
+
+    @Inject
+    StorageService storageService;
+
+    @Inject
+    NetworkDao networkDao;
+
+    @Inject
+    AccountDao accountDao;
+
+    @Inject
+    LockManager lockManager;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        if ( !MANAGE_NONRANCHER_CONTAIERS.get() ) {
+            return null;
+        }
+
+        ContainerEvent event = (ContainerEvent)state.getResource();
+        Map<String, Object> inspect = (Map<String, Object>)DataUtils.getFields(event).get(FIELD_DOCKER_INSPECT);
+
+        String rancherUuid = getRancherUuidLabel(inspect);
+        if ( rancherUuid != null ) {
+            return null;
+        }
+
+        scheduleInstance(event, inspect);
+
+        return null;
+    }
+
+    void scheduleInstance(final ContainerEvent event, final Map<String, Object> inspect) {
+        final Long accountId = event.getAccountId();
+        final String externalId = event.getExternalId();
+
+        lockManager.lock(new ContainerEventInstanceLock(accountId, externalId), new LockCallbackNoReturn() {
+            @Override
+            public void doWithLockNoResult() {
+                // TODO INDEX ON EXTERNAL ID
+                Instance instance = objectManager.findAny(Instance.class, INSTANCE.ACCOUNT_ID, accountId,
+                        INSTANCE.EXTERNAL_ID, externalId);
+                if ( instance == null ) {
+                    instance = objectManager.newRecord(Instance.class);
+                    instance.setKind(InstanceConstants.KIND_CONTAINER);
+                    instance.setAccountId(accountId);
+                    instance.setExternalId(externalId);
+                    setName(inspect, instance);
+                    setNetwork(accountId, inspect, instance);
+                    setImage(inspect, instance);
+                    setHost(event, instance);
+                    DataAccessor.fields(instance).withKey(InstanceConstants.FIELD_NATIVE_CONTAINER).set(Boolean.TRUE);
+                    instance = objectManager.create(instance);
+                }
+
+                if ( !InstanceConstants.STATE_RUNNING.equals(ObjectUtils.getState(instance)) &&
+                        !InstanceConstants.STATE_STARTING.equals(ObjectUtils.getState(instance))) {
+                    try {
+                        getObjectProcessManager().scheduleProcessInstance(InstanceConstants.PROCESS_CREATE, instance, makeData());
+                    } catch (ProcessCancelException e) {
+                        // TODO Figure out if i need this catch
+                    }
+                }
+            }
+        });
+    }
+
+    public static boolean isNativeDockerStart(ProcessState state) {
+        Boolean nativeDockerStart = DataAccessor.fromMap(state.getData()).withScope(ContainerEventCreate.class)
+                .withKey(InstanceConstants.DATA_NATIVE_DOCKER_START).as(Boolean.class);
```

Comment:
You can add "withDefault=false" condition to this statement. It will set the value to false in case its null